sklearn/preprocessing/tests/test_data.py

```diff
@@ -288,12 +288,101 @@ def test_polynomial_features_csr_X_dim_edges(deg, dim, interaction_only):
     assert_array_almost_equal(Xt_csr.A, Xt_dense)
 
 
-def test_standard_scaler_1d():
+def test_raises_value_error_if_sample_weights_greater_than_1d():
+    # Sample weights must be either scalar or 1D
+
+    n_sampless = [2, 3]
+    n_featuress = [3, 2]
+
+    for n_samples, n_features in zip(n_sampless, n_featuress):
+        X = rng.randn(n_samples, n_features)
+        y = rng.randn(n_samples)
+        sample_weights_OK = rng.randn(n_samples) ** 2 + 1
+        sample_weights_OK_1 = 1.
+        sample_weights_OK_2 = 2.
```

Comment:
I don't get why you test for 1 and 2 as you don't check the consequence on the estimated mean or var. Maybe just do 1?

Reply:
I would just remove the `sample_weights_OK*` smoke tests. Valid sample weights are already tested in the other tests. Let's focus this test on invalid sample weights (both wrong shape or wrong length).

Reply:
I removed most of those `sample_weight` tests except for the one testing `dtype` because it caused so much turmoil.
I also had to add `np.dot()` instead of `_safe_accumulator_op()` in `_incremental_weighted_mean_and_var` of `extmath.py` because it was again causing the `dtype` not found error.